replica/database.hh

```diff
@@ -1102,6 +1105,25 @@ public:
     void enable_off_strategy_trigger();
 
     compaction::table_state& as_table_state() const noexcept;
+
+    friend class compaction_group;
+};
+
+// Compaction group is a set of SSTables which are eligible to be compacted together.
+// By this definition, we can say:
+//      - A group contains SSTables from the same table and shard
+//      - Each group can be thought of an isolated LSM tree.
+//      - Time Window strategy has one group per window
+class compaction_group {
+    table& _t;
+    std::unique_ptr<table::table_state> _table_state;
+public:
+    compaction_group(table& t);
+
+    // Will stop ongoing compaction on behalf of this group, etc.
+    future<> stop();
```

Comment:
noexcept please.
exception in the `stop()` path are evil.

Reply:
stop() does flush a memtable, shouldn't we fail gracefully rather than aborting?

Reply:
Even if flush may fail gracefully, we should return that as an exceptional future.